pymeasure/display/curves.py

```diff
@@ -26,6 +26,7 @@
 
 import pyqtgraph as pg
 import numpy as np
+from matplotlib.cm import viridis  # JM: To make Neal happy ;)
```

Comment:
Nice! @ndr37, what do you think?

Currently matplotlib is considered an extra dependency, i.e. it is not required (see [setup.py#47](https://github.com/ralph-group/pymeasure/blob/60a2720486397dff8f6d5d85bcb211207545afbe/setup.py#L47)). We need to either promote it to a requirement or use it conditionally (using a try-catch on import). This is certainly open for debate. My first thought is to use it conditionally, so that we don't have to introduce more dependencies. Any thoughts?

Reply:
if this import is just for the colormap definition, it might be worth it to just hand-define viridis for pg use. There's a [PR open](https://github.com/pyqtgraph/pyqtgraph/pull/446) for including that into pyqtgraph itself, and a [related issue](https://github.com/pyqtgraph/pyqtgraph/issues/561), maybe you can pick a good way to do that from those.

Reply:
Thanks @bilderbuchi for finding the PR on pyqtgraph. It would be great to solve this upstream, but worst case, we can use can hand define it in the short term.

Reply:
I've made it optional and included a greyscale colormap to fall back on. I know that pyqtgraph has some builtin colormaps, but from what I read in the documentation it wasn't evident how to  use them without using an `ImageView` or `ImageWidget` which is why I am "hand-making" the image in `curves.ResulsImage`. Ultimately, switching to containing the image inside of one of the proper pyqtgraph image widgets would probably be best.

Reply:
ok, but if you just need the 2D array of colors (as evidenced by the greyscale fallback), why drag the mpl dependency in at all, if you can hardcode the values (which should not change) until this gets upstreamed, e.g. by `np.array(mp.cm.viridis.colors)` or somesuch?

Reply:
@jmittelstaedt @cjermain the mpl 2.0 colormaps are included in pyqtgraph 0.11 now.